### PR TITLE
[obs] install into plugin dir

### PR DESCRIPTION
**Pull request by arcnmx**
`/usr/share/obs/blahblah` is where obs expects to find the dylib, so put it there!

### CODE REVIEW

File: obs/CMakeLists.txt [hunk 1]
```diff
@@ -18,20 +18,27 @@ endif()
 option(ENABLE_BACKTRACE "Enable backtrace support on crash" ON)
 add_feature_info(ENABLE_BACKTRACE ENABLE_BACKTRACE "Backtrace support.")
 
-add_compile_options(
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
Comment (gnif) [inline obs/CMakeLists.txt:21]: Please split this up into separate commits, this has nothing to do with the installation path.

File: obs/CMakeLists.txt [hunk 2]
```diff
@@ -18,20 +18,27 @@ endif()
...
   "$<$<CONFIG:DEBUG>:-O0;-g3;-ggdb>"
 )
 
 set(EXE_FLAGS "-Wl,--gc-sections")
 set(CMAKE_C_STANDARD 11)
 
+if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+	set(OBS_PLUGIN_DIR 64bit)
+else()
+	set(OBS_PLUGIN_DIR 32bit)
```
Comment (gnif) [inline obs/CMakeLists.txt:39]: Do we even care about 32-bit, LG would be practically useless on such a system.
Also separate commit please

File: obs/CMakeLists.txt [hunk 3]
```diff
@@ -63,10 +59,15 @@ add_subdirectory("${PROJECT_TOP}/common"          "${CMAKE_BINARY_DIR}/common")
...
 add_library(looking-glass-obs SHARED ${SOURCES})
-target_link_libraries(looking-glass-obs
+target_compile_options(looking-glass-obs PRIVATE ${C_FLAGS})
+target_link_libraries(looking-glass-obs PRIVATE
 	${EXE_FLAGS}
 	lg_common
 	lgmp
 )
 
+install(TARGETS looking-glass-obs
```
Comment (gnif) [inline obs/CMakeLists.txt:69]: what about a local install into the local obs plugins folder for the user?

**Review by gnif — CHANGES_REQUESTED** [review]

File: obs/CMakeLists.txt (hunk 1 above)
Comment (arcnmx) [reply obs/CMakeLists.txt:21]: Unless I messed up these should be separate commits, though I can pull it out into another PR if you'd prefer.

File: obs/CMakeLists.txt (hunk 2 above)
Comment (arcnmx) [reply obs/CMakeLists.txt:39]: Dunno, but seems better than hard-coding 64bit. Separate commit from what, the `install()` or the cflags?

File: obs/CMakeLists.txt (hunk 3 above)
Comment (arcnmx) [reply obs/CMakeLists.txt:69]: Good point, that might make sense as an option... `${CMAKE_INSTALL_DATADIR}/obs/obs-plugins/` becomes `$ENV{HOME}/.config/obs-studio/plugins/` I think?

File: obs/CMakeLists.txt (hunk 1 above)
Comment (gnif) [reply obs/CMakeLists.txt]: No, this is part of the one commit, please split it up.

File: obs/CMakeLists.txt (hunk 2 above)
Comment (gnif) [reply obs/CMakeLists.txt:38]: part of the install changes sorry, and yeah, keep this, it's fine :)

File: obs/CMakeLists.txt (hunk 1 above)
Comment (arcnmx) [reply obs/CMakeLists.txt]: I just removed the second commit from this PR, I'll continue to experiment with my local install in [another branch](https://github.com/arcnmx/LookingGlass/compare/obs-cmake...arcnmx:obs-cmake2). None of it is all that relevant anyway, I was just trying to clean up a bit and tie the flags to targets to make it easier to customize.
